lib/src/ruby/module.cc

```diff
@@ -112,20 +113,47 @@ extern "C" {
      */
     void LIBFACTER_EXPORT Init_libfacter()
     {
-        setup_logging(boost::nowide::cerr);
-        set_level(log_level::warning);
+        bool logging_init_failed = false;
+        string logging_init_error_msg;
+        try {
+            facter::logging::setup_logging(boost::nowide::cerr);
+            set_level(log_level::warning);
+        } catch(facter::logging::locale_error const& e) {
+            logging_init_failed = true;
+            logging_init_error_msg = e.what();
+        }
 
         // Initialize ruby
+        api* ruby = nullptr;
         try {
-            auto& ruby = api::instance();
-            ruby.initialize();
+            ruby = &api::instance();
```

Comment:
Going to let @peterhuene have a look at this, he spent more time on Ruby init than I. However, it seems highly unlikely api::instance would fail when initializing when loaded by Ruby, so this all looks reasonable.